ui/components/compose.jsx

```diff
@@ -77,36 +45,72 @@ module.exports = function(core, config, store) {
 				to: nav.room,
 				from: store.get("user"),
 				text: text,
-				time: new Date().getTime(),
+				time: Date.now(),
 				thread: nav.thread
 			});
 
 			this.replaceState(this.getInitialState());
+		},
+
+		getPlaceHolder: function() {
+			return ("Reply as " + appUtils.formatUserName(store.get("user")));
+		},
+
+		getMessageText: function(input) {
+			var currentText = store.get("nav", "currentText"),
+				textObj = store.get("indexes", "textsById", currentText),
+				msg = input || "",
+				user, nick, atStart;
+
+			if (textObj) {
+				nick = appUtils.formatUserName(textObj.from);
+				user = appUtils.formatUserName(store.get("user"));
 
-			// FIXME: figure out why replaceState is not working
-			composeBox.innerText = composeBox.textContent = this.getInitialState().userInput;
+				if (/^@\S+[\s+{1}]?/.test(msg)) {
```

Comment:
@aravindet `/^@\S+[\s+{1}]?/` is the same as `/^@\S+\s?/`or `/^@\S+\s{1}/` (I'll change it).